dvc/ignore.py

```diff
@@ -357,20 +367,32 @@ def _outside_repo(self, path):
             return True
         return False
 
-    def check_ignore(self, target):
+    def check_ignore(self, target) -> CheckIgnoreResult:
         # NOTE: can only be used in `dvc check-ignore`, see
         # https://github.com/iterative/dvc/issues/5046
+        from dvc.exceptions import DvcException
+
         full_target = os.path.abspath(target)
-        if not self._outside_repo(full_target):
-            dirname, basename = os.path.split(os.path.normpath(full_target))
-            pattern = self._get_trie_pattern(dirname)
-            if pattern:
-                matches = pattern.matches(
-                    dirname, basename, os.path.isdir(full_target), True,
-                )
+        if self._outside_repo(full_target):
+            raise DvcException(
```

Comment:
This is a problem here that, `check-ignore` is also used in `output validation`, although only for message show.
After this merged, these two exceptions and error message would also be thrown to 

https://github.com/iterative/dvc/blob/31439754377d829da6babffcc720dd696554d8ab/dvc/output/base.py#L626-L627

And the error message here would be shown.

Reply:
Error message from cls.IsIgnoredError
```
f"Path '{match.file}' is ignored by\n{lines}")
```

Reply:
Maybe we should not use `fatal` which follows `Git` here to make messages more adapt to our own.

Reply:
Agreed, let's get rid of `fatal` prefix.